src/d/d_player_mdl.cpp

```diff
@@ -0,0 +1,1959 @@
+    {           "LiftS_UtoD",   7,   0, 57, 57, 57},
+    {            "LiftLWalk",   7,   0, 57, 57,  2},
+    {          "MPGuardPose", 255, 255, 57, 57, 57},
+    {             "RockRide", 255,  56, 12, 57, 12},
+    {           "BinDrinkSt", 255,   5, 13, 13, 13},
+    {             "BinDrink", 255,   5, 14, 57, 14},
+    {           "BinDrinkEd", 255,   5, 15, 57, 15},
+    {             "HarpPose",   7,   6,  1,  0,  0},
+    {           "HarpPlayGo",   7,   1,  1,  0,  0},
+    {         "HarpPlayCome",   7,   0,  1,  0,  0},
+    {              "BinHold", 255,   5, 57, 57, 57},
+    {            "BinShield",   9,   5, 24, 24, 24},
+};
+#pragma pop
+
+const s32 daPlayerModelBase_c::sUnkDuration = 0x2400;
+
+const u8 daPlayerModelBase_c::sShieldDurabilities[10] = {16, 24, 32, 24, 36, 48, 12, 16, 20, 48};
+const u8 daPlayerModelBase_c::sShieldRegenTimes[10] = {0, 0, 0, 0, 0, 0, 90, 90, 90, 0};
+
+/**
+ * Watch the player's hand material indices and change the ResMat
+ * so that the right hand mats are visible and all others are hidden.
+ */
+void daPlBaseScnObjCallback_c::ExecCallback_DRAW_OPA(
+    nw4r::g3d::ScnObj::Timing timing, nw4r::g3d::ScnObj *pObj, u32 param, void *pInfo
+) {
+    if (timing != nw4r::g3d::ScnObj::CALLBACK_TIMING_C) {
+        nw4r::g3d::ScnMdl *pMdl = nw4r::g3d::ScnObj::DynamicCast<nw4r::g3d::ScnMdl>(pObj);
+        nw4r::g3d::ResMdl mdl = pMdl->GetResMdl();
+        nw4r::g3d::ResShp shp(nullptr);
+        for (int i = 0; i < 2; i++) {
+            u8 defaultMat = sSavedHandMats[i];
+            // TODO: Is there a way to avoid this cast?
+            u8 playerHandMat = static_cast<daPlayerModelBase_c *>(field_0x0C)->getHandMat(i);
+            if (playerHandMat != defaultMat) {
+                if (defaultMat != 0xFE) {
+                    shp = mdl.GetResShp(sHandMats[defaultMat]);
+                    shp.SetVisibility(false);
+                    shp.DCStore(false);
+                }
+                if (playerHandMat != 0xFE) {
+                    shp = mdl.GetResShp(sHandMats[playerHandMat]);
+                    shp.SetVisibility(true);
+                    shp.DCStore(false);
+                }
+                sSavedHandMats[i] = playerHandMat;
+            }
+        }
+    }
+}
+
+void daPlBaseScnObjCallback_c::ExecCallback_CALC_MAT(
+    nw4r::g3d::ScnObj::Timing timing, nw4r::g3d::ScnObj *pObj, u32 param, void *pInfo
+) {
+    if (timing != nw4r::g3d::ScnObj::CALLBACK_TIMING_C) {
+        return;
+    }
+    dScnCallback_c::ExecCallback_CALC_MAT(timing, pObj, param, pInfo);
+}
+
+void daPlBaseMdlCallback_c::ExecCallbackA(
+    nw4r::g3d::ChrAnmResult *result, nw4r::g3d::ResMdl mdl, nw4r::g3d::FuncObjCalcWorld *calcWorld
+) {
+    u16 nodeId = calcWorld->GetCallbackNodeID();
+    // Only run the full callback for certain specified nodes - TODO which ones?
+    if ((nodeId >= mNodeIdMin && nodeId < mNodeIdMax) || (mNodeIdMin == 16 && nodeId == 0)) {
```

Comment:
this could make sense for link's main body? 16 is waist down and 0 is center

Reply:
I'm not sure why it checks mNodeIdMin though; this file sets the range to 0-0xFFFF